cfme/tests/automate/test_common_methods.py

```diff
@@ -180,3 +180,89 @@ def test_miq_password_decrypt(klass):
         execute_methods=True,
     )
     assert result.validate()
+
+
+@pytest.fixture(scope='module')
+def original_class(domain):
+    # Take the 'Request' class and copy it for own purpose.
+    domain.parent.instantiate(name="ManageIQ").namespaces.instantiate(
+        name="System"
+    ).classes.instantiate(name="Request").copy_to(domain.name)
+    klass = domain.namespaces.instantiate(name="System").classes.instantiate(name="Request")
+    return klass
+
+
+@pytest.mark.tier(1)
+@pytest.mark.meta(automates=[1700524])
+@pytest.mark.ignore_stream("5.10")
+def test_service_retirement_from_automate_method(request, generic_catalog_item, original_class):
+    """
+    Bugzilla:
+        1700524
+
+    Polarion:
+        assignee: ghubale
+        initialEstimate: 1/8h
+        caseposneg: positive
+        startsin: 5.11
+        casecomponent: Automate
+        testSteps:
+            1. Create service catalog item and order
+            2. Create a writeable domain and copy ManageIQ/System/Request to this domain
+            3. Create retire_automation_service instance and set meth5 to retire_automation_service.
+            4. Create retire_automation_service method with sample code given below:
+               > service = $evm.root['service']
+               > $evm.log(:info, "create_retire_request for  service #{service}")
+               > request = $evm.execute(:create_retire_request, service)
+               > $evm.log(:info, "Create request for create_retire_request #{request}")
+            5. Execute this method using simulation
+    """
+    # Ordering catalog item and deleting request once service has been reached to 'Finished' state
+    service_request = generic_catalog_item.appliance.rest_api.collections.service_templates.get(
+        name=generic_catalog_item.name
+    ).action.order()
+    request.addfinalizer(lambda: service_request.action.delete())
+    wait_for(lambda: service_request.request_state == "finished", fail_func=service_request.reload,
+             timeout=180, delay=10)
+
+    # Ruby code to execute create_retire_request
+    script = (
+        "service = $evm.root['service']\n"
+        "$evm.log(:info, 'create_retire_request for service #{service}')\n"
+        "request = $evm.execute(:create_retire_request, service)\n"
+        "$evm.log(:info, 'Create request for create_retire_request #{request}')"
+    )
+
+    # Adding method - send_email for sending mails
+    method = original_class.methods.create(
+        name=fauxfactory.gen_alphanumeric(),
+        display_name=fauxfactory.gen_alphanumeric(),
+        location='inline',
+        script=script)
+
+    # Adding instance to call automate method - send_email
+    instance = original_class.instances.create(
```

Comment:
Any cleanup necessary for this instance?

Reply:
- Actually there is no cleanup required for instance as it is child of klass/namespace/domain. Instance/method is automatically gets deleted once we delete klass/namespace/domain.
- It will consume more time as it will delete method - instance - klass- namespace - domain
- But I am getting these question again angain hence adding cleanup.

Reply:
Apologies for the repeated question! Maybe to avoid this question in the future you could put an inline comment explaining why no cleanup is necessary? It would certainly help jog my memory :smile:

Reply:
- I am not saying repeated comments from you only but it is obvious that all are always asking about that as I am creating something there(like method/instance)
- But thanks for inline comment suggestion, I will start adding in coming PRs